### DISCUSSION TITLE
Custom shortcode doesn't use codeblock padding

### DISCUSSION THREAD

**Discussion by ghost** [imfing/hextra#26] [Q&A] [answered]
I'm using the below shortcode to fetch data from a git repository:

```
{{ $file := .Get 0 }}
{{ with resources.GetRemote $file }}
  {{ with .Err }}
    {{ errorf "%s" . }}
  {{ else }}
    {{ $lang := path.Ext $file | strings.TrimPrefix "." }}
    {{ highlight .Content $lang }}
  {{ end }}
{{ else }}
  {{ errorf "Unable to get remote resource." }}
{{ end }}
```

When I use this shortcode, there is no padding on the left and right.

[This](https://codeberg.org/mishra/ferris.rs) is the repository in question, and [this](https://codeberg.org/mishra/ferris.rs/src/branch/main/content/exercises/annalyns-infiltration.md) is the file where I using the shortcode.

Steps to reproduce:
+ Start Hugo server
+ Navigate to Exercises > Annalyn's Infiltration
+ Scroll down

How do I ensure that the code which is fetched has the same padding as the code which is already part of the same repository?

**Comment by imfing** [comment, marked as answer]
Need to add the wrapper around code block:
https://github.com/imfing/hextra/blob/8e8f7f23c955bd31f9faba501a2121d74e83793c/layouts/_default/_markup/render-codeblock.html#L6

This can be done via either in your shortcode:
```
    {{ $lang := path.Ext $file | strings.TrimPrefix "." }}
    <div class="code-block relative mt-6 first:mt-0 group/code">
      {{ highlight .Content $lang }}
    </div>
```

or in the markdown:
```
<div class="code-block relative mt-6 first:mt-0 group/code">
{{< codeberg_source "https://codeberg.org/mishra/rustacean/raw/branch/main/src/getting_started/annalyns_infiltration.rs" >}}
</div>
```
<img width="1321" alt="image" src="https://github.com/imfing/hextra/assets/5097752/f058944c-80c8-4e59-9920-66c97b6b832b">

A caveat is that this code block would not have the copy button

**Reply by ghost** [reply]
I updated the shortcode

**Reply by imfing** [reply]
Fabulous!
